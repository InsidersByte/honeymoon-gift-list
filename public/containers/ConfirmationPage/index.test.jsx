import React from 'react';
import { shallow, mount } from 'enzyme';
import { ConfirmationPage } from './';

const loadGiftSet = jest.fn();

const props = {
    loading: true,
    giftSet: {
        paymentMethod: 'method',
        paypalLink: 'http://paypal.me',
    },
    params: { giftSetId: 1 },
    actions: { loadGiftSet },
};

describe('ConfirmationPage', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<ConfirmationPage {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should initialise state', () => {
        const wrapper = shallow(<ConfirmationPage />);

        expect(wrapper.state()).toEqual({ linkClicked: false });
    });

    it('should load giftSet on componentDidMount', () => {
        mount(<ConfirmationPage {...props} />);

        expect(loadGiftSet).toHaveBeenCalledWith(1);
    });

    it('should render a <Confirmation /> component', () => {
        const wrapper = shallow(<ConfirmationPage />);

        expect(wrapper.find('Confirmation').length).toBe(1);
    });

    it('should set <Confirmation /> props', () => {
        const wrapper = shallow(<ConfirmationPage {...props} />);

        const confirmation = wrapper.find('Confirmation');

        expect(confirmation.prop('giftSet')).toBe(props.giftSet);
        expect(confirmation.prop('loading')).toBe(props.loading);
        expect(confirmation.prop('linkClicked')).toBe(false);
        expect(confirmation.prop('onLinkClicked')).toBeInstanceOf(Function);
    });

    it('should set linkClicked to true when link is clicked', () => {
        const wrapper = shallow(<ConfirmationPage />);

        expect(wrapper.state()).toEqual({ linkClicked: false });
        wrapper.instance().onLinkClicked();
        expect(wrapper.state()).toEqual({ linkClicked: true });
        wrapper.instance().onLinkClicked();
        expect(wrapper.state()).toEqual({ linkClicked: true });
    });
});
