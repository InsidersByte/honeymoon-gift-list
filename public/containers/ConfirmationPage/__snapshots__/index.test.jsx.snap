// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationPage should render correctly 1`] = `
<Confirmation
  giftSet={
    Object {
      "paymentMethod": "method",
      "paypalLink": "http://paypal.me",
    }
  }
  linkClicked={false}
  loading={true}
  onLinkClicked={[Function]}
/>
`;
